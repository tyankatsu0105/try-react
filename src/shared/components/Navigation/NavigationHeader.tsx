import React, { FunctionComponent } from 'react';
import clsx from 'clsx';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { AppBar, Toolbar, IconButton, Typography } from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';

const drawerWidth = 8 * 30;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    navigationHeader: {
      zIndex: theme.zIndex.drawer + 1,
      transition: theme.transitions.create(['width', 'margin'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
    },
    'navigationHeader-shift': {
      marginLeft: drawerWidth,
      width: `calc(100% - ${drawerWidth}px)`,
      transition: theme.transitions.create(['width', 'margin'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.enteringScreen,
      }),
    },
    'navigationHeader-MenuButton': {
      marginRight: 36,
    },
    'navigationHeader-MenuButton-hide': {
      display: 'none',
    },
  })
);

interface Props {
  open: boolean;
  handleDrawerOpen: () => void;
}

export const NavigationHeader: FunctionComponent<Props> = (props) => {
  const classes = useStyles();

  const { open, handleDrawerOpen } = props;

  return (
    <AppBar
      className={clsx(classes['navigationHeader'], {
        [classes['navigationHeader-shift']]: open,
      })}
    >
      <Toolbar>
        <IconButton
          color="inherit"
          aria-label="open drawer"
          onClick={handleDrawerOpen}
          edge="start"
          className={clsx(classes['navigationHeader-MenuButton'], {
            [classes['navigationHeader-MenuButton-hide']]: open,
          })}
        >
          <MenuIcon />
        </IconButton>
        <Typography variant="h6" noWrap>
          Mini variant drawer
        </Typography>
      </Toolbar>
    </AppBar>
  );
};
