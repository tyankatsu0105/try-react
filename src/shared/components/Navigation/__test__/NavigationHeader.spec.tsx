import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { NavigationHeader } from '../NavigationHeader';

describe('NavigationHeader', () => {
  it('snapshot', () => {
    const props = {
      open: true,
      handleDrawerOpen: jest.fn(),
    };
    const { baseElement } = render(<NavigationHeader {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
