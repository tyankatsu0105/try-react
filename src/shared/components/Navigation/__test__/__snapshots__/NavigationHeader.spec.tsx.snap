// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationHeader snapshot 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-navigationHeader-1 makeStyles-navigationHeader-shift-2 MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-navigationHeader-MenuButton-3 makeStyles-navigationHeader-MenuButton-hide-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
        >
          Mini variant drawer
        </h6>
      </div>
    </header>
  </div>
</body>
`;
